packages/utils/src/PromiseUtils.ts

```diff
@@ -18,10 +18,13 @@ export class PromiseUtils {
   ): CancelablePromise<T> {
     let hasCanceled = false;
     let resolved: T | undefined;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let rejectFn: (val?: any) => void;
 
     const wrappedPromise = new Promise((resolve, reject) => {
-      Promise.resolve(promise).then(
-        val => {
+      rejectFn = reject;
+      Promise.resolve(promise)
+        .then(val => {
```

Comment:
With these changes, `reject` is called right after `hasCanceled` is set to true. If `hasCanceled` is true, there's no reason to call `reject` in those scenarios. It will just get swallowed up since the promise is already fulfilled, but it is unnecessary and could be removed (lines 32, and the true case on line 39).